### ISSUE TITLE
tf.signal.rfft returns non-zero / non-finite values for zero-length input axis with positive explicit fft_length

### ISSUE THREAD

**Issue by dingjiawei65** [tensorflow/tensorflow#126898]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

 Windows

### Python version

3.11

### Current behavior?

tf.signal.rfft(tf.zeros((2, 2, 0), dtype=tf.float32), fft_length=[16])`
should return a `complex64` tensor with shape `[2, 2, 9]`. Since the input is
empty along the transform axis and the positive explicit FFT length pads with
zeros, all output values should be finite zeros.

An alternative acceptable behavior would be a clear validation error if
TensorFlow intentionally does not support this boundary case. Returning
non-zero or non-finite values appears incorrect.

The result shape is `[2, 2, 9]`, but some runs contain non-zero and/or
non-finite values, including `nan+nanj`. The failure is intermittent within a
single process, which looks like uninitialized output memory.

### Standalone code to reproduce the issue

```shell
import os
import platform
import tensorflow as tf

TRIALS = 100
EXPECTED_SHAPE = [2, 2, 9]

print("python:", platform.python_version())
print("platform:", platform.platform())
print("tensorflow:", tf.__version__)
print("TF_ENABLE_ONEDNN_OPTS:", os.environ.get("TF_ENABLE_ONEDNN_OPTS"))

bad_shape = 0
nonfinite = 0
nonzero = 0
first_bad = None

for trial in range(TRIALS):
    y = tf.signal.rfft(
        tf.zeros((2, 2, 0), dtype=tf.float32),
        fft_length=[16],
    )

    real = tf.math.real(y)
    imag = tf.math.imag(y)
    is_finite = bool(
        tf.reduce_all(
            tf.logical_and(tf.math.is_finite(real), tf.math.is_finite(imag))
        ).numpy()
    )
    nonzero_count = int(
        (tf.math.count_nonzero(real) + tf.math.count_nonzero(imag)).numpy()
    )

    if y.shape.as_list() != EXPECTED_SHAPE:
        bad_shape += 1
    if not is_finite:
        nonfinite += 1
    if nonzero_count:
        nonzero += 1
    if first_bad is None and (not is_finite or nonzero_count):
        first_bad = {
            "trial": trial,
            "shape": y.shape.as_list(),
            "first_values": y.numpy().reshape(-1)[:10].tolist(),
        }

print(
    {
        "trials": TRIALS,
        "expected_shape": EXPECTED_SHAPE,
        "bad_shape": bad_shape,
        "nonfinite": nonfinite,
        "nonzero": nonzero,
        "first_bad": first_bad,
    }
)
```

### Relevant log output

```shell
{'trials': 100, 'expected_shape': [2, 2, 9], 'bad_shape': 0, 'nonfinite': 7, 'nonzero': 7, 'first_bad': {'trial': 1, 'shape': [2, 2, 9], 'first_values': [0j, 0j, 0j, 0j, (-2.181645667963854e+30+4.5914945482066956e-41j), 0j, (1.7796490496925177e-43+0j), (2.677369972350537e+36+8.169570047013684e-43j), (2.677537302229767e+36+8.169570047013684e-43j), (2.6777198439162e+36+8.169570047013684e-43j)]}}
```

**Comment by vishwakt** [comment]
I have a solution for this and can make a fix

**Comment by Kayyuri** [comment]
Hi @dingjiawei65,  Thank you for raising your concern here!
A fix for this issue has been proposed in this [PR](https://github.com/tensorflow/tensorflow/pull/126909). Once it is merged, this issue will be resolved.

Hi @vishwakt,
Thank you for raising the PR to address this issue. We truly appreciate your initiative and the effort in contributing the fix.
